### ISSUE TITLE
Is this project dead?

### ISSUE THREAD

**Issue by foobar1234** [RequestPolicy/requestpolicy#407]
Last activity from author was half a year ago, the website https://www.requestpolicy.com/ has some non-working links, the official support mail address is dead as well (got a mailer daemon answer). And all that while stable 0.5.x has some major drawbacks with latest FF (some requests are blocked, but not allowable). 1.0.x series is in non-activity-beta about a year now without any updates, even though the mailing lists and issues are full of bug reports.

So, is this project dead? Are there similar alternatives?

**Comment by ghost** [comment]
The project does seem dead. The author seems busy with his project as mentioned on the official homepage.

Unfortunately there does not seem to be any alternative to RequestPolicy. 

The best option would be for Justin to hand over the project to a new maintainer (if someone _is_ willing in the first place). Another option would be to fork this and continue development on the fork.

**Comment by cryptonot** [comment]
From his homepage:
"...
My primary open source project is my Firefox extension RequestPolicy. I began working on RequestPolicy in 2008 and it's now used by tens of thousands of people. As of late 2011, it's also my primary research project. So, I'm _supposed_ to be working on it. Just thinking about that is enough to make me feel like one of the luckiest people alive. How many people get paid to work on something they're passionate about? (I guess the luckier ones get paid more than a stipend to do so.) 
..."

This doesn't sound like he dropped it. And the last commit was 4 months ago, according to the commit history there have been longer gaps in development. But I'm also a bit disappointed he didn't  update his Request Policy 1.0 beta in a year (https://github.com/RequestPolicy/requestpolicy/tree/dev-1.0).
I guess he's just rather busy right now.

**Comment by JordanMilne** [comment]
I think it might be. I sent an email to Justin two months ago with the patch in my pull request, but didn't get a response. Unfortunately, I don't think there's any chance of getting the patch pushed to addons.mozilla.org without his blessing.

**Comment by leaumar** [comment]
Please Justin, don't leave us hanging :(

**Comment by dfc** [comment]
Did you donate to Justin: https://www.requestpolicy.com/support-requestpolicy.html

Otherwise it just seems like you are leaving him hanging. :`(

**Comment by leaumar** [comment]
Dcf, I'm an add-on dev and developer of a million other things myself, all of which I have to maintain. I know the pain of it, but I do a lot without donations too and asking for at least a sign of life is not something that needs paying for

**Comment by bonanza123** [comment]
What do you think can we do to convince him or a successor to continue/start mainting this great project or a fork of it? Maybe starting a call to collect some money?

**Comment by leaumar** [comment]
I honestly don't know. For me personally, the main motivation for my projects is knowing that enough people use them, seeing the download count go up steadily and getting feedback from users... I think RP has plenty of that, so I can only think of money or a big surprise publicity stunt for Justin as other motivators.

**Comment by myrdd** [comment]
> download count go up steadily and getting feedback from users... I think RP has plenty of that

That's right. I also made several plans on how to make this extension better, although I haven't implemented those ideas so far. We could at least try to make a community project, e.g. with an irc channel to talk.

**Comment by leaumar** [comment]
I doubt such a thing would get used a lot... RP is not exactly a community thing like a game.

The best I can think of is just updating the source ourselves with a handful of programmers willing to spend some time on it. We CAN make our own commits to this, right? Or will we need to make a fork?

**Comment by myrdd** [comment]
Maybe you're right. But at least, if there are several persons interested in improving this, we could at least exchange experiences with each other.
Justin would need to allow us to do changes, and this is not the case at the moment, so I would say we start with a folk (or several ones). The problem is, the issues will stay on the original project. Is there a solution to this? – of course, it would be great if we could work on this project directly. Can we get Justin to that?

**Comment by bonanza123** [comment]
What about creating a fork of this great project? Would that be legally possible? I mean I'm not an expert neither on programming nor on the licenses, the only thing I could do is donating some beers. But maybe there are some more (useless) people like me :-D

**Comment by myrdd** [comment]
There are already some forks of this project (see ![here](https://github.com/RequestPolicy/requestpolicy/network)) although there aren't big changes. I'm willing to work on this, but I don't know how far I get as this is my first firefox extension I work on. :)

**Comment by bonanza123** [comment]
Thanks for the hint. But the Network-link you provided doesnt show all forks right? I.e. yours is missing. Maybe there is also the possibility of contacting the other guys who forked the project to collaboratively work on a single fork?

**Comment by myrdd** [comment]
That's right, it shows only forks which are public and which already have at least one commit. In my case, I didn't make any commit so far.
To me it seems that futpib's dev-1.0 branch is the most interesting one atm. He took parts of swick's commits and made some more commits.

**Comment by foobar13372** [comment]
https://www.requestpolicy.com/ has an outdated certificate for over one week now. Surely this project is dead. Any alive forks with significant work pu into?

**Comment by nodiscc** [comment]
Is someone willing to start having a look at the changes proposed in the 34 current forks on github and try to merge some of them? Transfering the repo to a github organization would also be a good thing? @jsamuel ping, we need your input

**Comment by myrdd** [comment]
Yes, I like the idea of making RP a community project / github organization. I would support it for sure. I've been giving a lot of thought to RP already, although I didn't program anything so far (sorry if you expected that from my above posts).
Yes, it would be nice if @jsamuel gave his two cents, but as he didn't show up for quite some time already, we might work together on a fork (by adding interested persons as collaborators to a fork).
BTW, I don't think this project is dead just because the website certificate outdated. There's quite some activity here on github :)

**Comment by nodiscc** [comment]
@myrdd nice to hear from you, so what's the first step? I think creating another Github organization/repo would be good, then maybe we can start copying the most important issues there to see what needs work (I've just hit #395 aka #313 aka #386 aka #372 aka #362 aka #410 - yes they are really the same issue, reported multiple times and not cleaned up).

**Comment by myrdd** [comment]
Hi nodiscc! Yes, the isues need to be structured somehow.

I can see you're doing many contributions, so you're more familiar with github. IMO it needn't be a new organization, we could even start with a user repository, like mine https://github.com/myrdd/rsnapshot (I didn't make any changes to this fork yet.) We could then later move to an organization as soon as we got some steps further.
In case you prefer to create an organization, we should think of a name. Maybe RequestPolicyC, "C" meaning "community"(?).

Isn't it possible to migrate/copy issues to a fork?

**Comment by nodiscc** [comment]
> Isn't it possible to migrate/copy issues to a fork?

Officially no, there's this script though: https://github.com/IQAndreas/github-issues-import

A simple user repository looks ok

**Comment by nodiscc** [comment]
Working issues copy: https://github.com/requestpolicy-test/requestpolicy

**Comment by myrdd** [comment]
finally I've done some development. See https://github.com/myrdd/requestpolicy/commit/5cca550568212cebd1f30f51ec94cc861d9038b2
@nodiscc great, thank you! what do you think about renaming the organization to `RequestPolicy-alternate`? Could you please allow me to manage issues (creating/assigning labels, closing issues etc.)? I'd like to organize them.

**Comment by nodiscc** [comment]
@myrdd sorry because I didn't see your reply, great work at https://github.com/RequestPolicyContinued/requestpolicy/commits/dev-1.0

I've added you as contributor at https://github.com/requestpolicy-community/requestpolicy, is the project rename OK? You might want to move issues from https://github.com/RequestPolicyContinued/requestpolicy/issues there.

Please take a look at https://github.com/RequestPolicyContinued/requestpolicy/commit/d7f4da7fbfbe2abbc82f372e61b9ee848fc60870#commitcomment-7408303 also; thanks again!

**Comment by myrdd** [comment]
@nodiscc glad to hear from you. yes, we already did some work on the fork. Thanks for adding me as a contributor and renaming the organization. Now we have to think about how we'll merge the organizations:

First, about the name, I would prefer RequestPolicyContinued for now, not only because we use it already for some days, but also to emphasize the fact there hasn't been development for some time. What do you think?

Second, about issues. I think it would be good to have all the issues in the new repository. I've already categorized almost all issues on my pc, so it would be easy to remove duplicates etc.

So how do we merge. Is it possible to import issues from https://github.com/RequestPolicyContinued/requestpolicy/issues, in addition to those already there? Pulling the commits made to RPC is easy. So then we only need to remove https://github.com/RequestPolicyContinued/requestpolicy and rename _requestpolicy-test_ to _RequestPolicyContinued_ – is that possible?

**Comment by nodiscc** [comment]
> is that possible

Yes completely, please push your commits to https://github.com/requestpolicy-community/requestpolicy

I'll migrate the issues from https://github.com/RequestPolicyContinued/requestpolicy/issues there, and when you delete your organization I'll rename _requestpolicy-community_ to _RequestPolicyContinued _

**Comment by myrdd** [comment]
great! ok, I've pushed to _requestpolicy-community_ and renamed _RequestPolicyContinued_ to _RequestPolicyContinuedTmp_.

**Comment by nodiscc** [comment]
_requestpolicy-community_ renamed to _RequestPolicyContinued_

Moving your issues there as soon as I have time.

**Comment by myrdd** [comment]
great! Thanks for your effort!

**Comment by bonanza123** [comment]
Is https://sslsites.de/requestpolicy.256k.de/ now the 'new' release website? And does the plugin from there work with the 'old' 1.0-beta? Or how to keep my old settings?

**Comment by myrdd** [comment]
If you wouly like to test the "continued" beta, you can either compile it by yourself _or_ you use the xpi file offered on that site. All changes I've made so far were fully backwards-compatible, so you can even switch back to Justin's 1.0 version if you wish to.

The URL itself though might change – we will move the xpi and update.rdf to github if possible. See the discussion here: RequestPolicyContinued#445

**Comment by bonanza123** [comment]
@myrdd thanks! I'm so glad that you guys are continuing this great project! Btw. are there any plans to open some bounty for keeping you guys motivated? I'm sure there are a lot of people who are willing to show their acknowledgment with some beers ;-).

**Comment by myrdd** [comment]
@bonanza123 thanks for your positive feedback! hm yes, that might be a good idea. :) maybe via [flattr](https://flattr.com)?

**Comment by nodiscc** [comment]
[Gittip](https://www.gittip.com/about/) is another alternative, it allows sending money to someone based on their Github username. Payments outside the U.S. not available yet though.

Combine this with a [contributors list widget](https://blossom.github.io/contributors/) on the homepage and we're good to go.

Flattr is a good idea but we'd better allow donations to individual contributors (else how are donations to the RP flattr account redistributed?) Flattr fees are also generally higher (10%).

Not saying gittip is the answer. Maybe this should be filed as a new issue on https://github.com/RequestPolicyContinued/requestpolicy/issues ?

**Comment by bonanza123** [comment]
What about the PayPal fees are they comparable? And what about the anonymity of these provider?

**Comment by myrdd** [comment]
gittip sounds interesting, although it would be nice no to depend on a company (but digital cash in euros/usd doesn't exist yet.) However, I like the idea of being able to send a tip to anyone in the project! For that, a contributor-list is a good idea.

Yes, we could discuss this in a new issue. You want to create one?

**Comment by nodiscc** [comment]
@myrdd I see you deleted the RequestPolicyTmp organziation before I could move your issues to RequestPolicyContinued. Sorry about that.

> Yes, we could discuss this in a new issue. You want to create one?

I personnally don't want to receive donations as I didn't do much work on this project. I've set up a draft for the new homepage with a widget that shows Github contributors, but it misses most persons that should receive credit for RP (a list at https://www.requestpolicy.com/about.html). I'll probably end up adding everyone to an `AUTHORS` file in the repository and linking to it from the homepage.

Is this ok?

**Comment by myrdd** [comment]
> @myrdd I see you deleted the RequestPolicyTmp organziation before I could move your issues to RequestPolicyContinued. Sorry about that.

[I didn't](https://github.com/RequestPolicyContinuedTmp/requestpolicy). If importing issues is an easy task for you, thank you! Otherwise we can move issues by hand, they're just a few. By the way, I really appreciate your contributions! :)

> I'll probably end up adding everyone to an AUTHORS file in the repository and linking to it from the homepage.
> 
> Is this ok?

In any case I'd mention everyone from the _about_ page. More I can say as soon as I see a draft of the website.

**Comment by nodiscc** [comment]
I moved the issues from https://github.com/RequestPolicyContinuedTmp/requestpolicy/issues to https://github.com/RequestPolicyContinued/requestpolicy/issues. Ongoing work on the project website at https://github.com/RequestPolicyContinued/requestpolicy/issues/445

**Comment by myrdd** [comment]
Thank you @nodiscc !
